src/staking/manager.ts

```diff
@@ -0,0 +1,660 @@
+import { networks, Psbt, Transaction } from "bitcoinjs-lib";
+import { StakingParams, VersionedStakingParams } from "../types/params";
+import { UTXO } from "../types";
+import { StakerInfo, Staking } from ".";
+import { fromBech32 } from "@cosmjs/encoding";
+import {
+  btccheckpoint,
+  btcstaking,
+  btcstakingtx,
+} from "@babylonlabs-io/babylon-proto-ts";
+import {
+  BTCSigType,
+  ProofOfPossessionBTC,
+} from "@babylonlabs-io/babylon-proto-ts/dist/generated/babylon/btcstaking/v1/pop";
+import { BBN_REGISTRY_TYPE_URLS } from "../constants/bbnRegistry";
+import { createCovenantWitness } from "./transactions";
+import { getBbnParamByBtcHeight, getBbnParamByVersion } from "../utils/staking/param";
+import { reverseBuffer, uint8ArrayToHex } from "../utils";
+
+export interface BtcProvider {
+  // Sign a PSBT
+  signPsbt(psbtHex: string): Promise<string>;
+  // Sign a message using the ECDSA type
+  signMessage: (message: string, type: "ecdsa") => Promise<string>;
+  // Get the staker info
+  getStakerInfo(): Promise<StakerInfo>;
+  // Get the available UTXOs for staking
+  getUTXOs(): Promise<UTXO[]>;
+}
+
+export interface BbnProvider {
+  // Get the Babylon BTC tip height. Note the there might be a delay between
+  // the BTC tip height and the Babylon BTC tip height. You should not use the
+  // BTC tip height to determine the BBN param.
+  getBabylonBtcTipHeight(): Promise<number>;
+  // Get the Babylon Chain address
+  getBbnAddress(): Promise<string>;
+  // Sign a babylon transaction.
+  signTransaction: <T extends object>(msg: {
+    typeUrl: string;
+    value: T;
+  }) => Promise<Uint8Array>
+}
+
+interface StakingInputs {
+  finalityProviderPkNoCoordHex: string;
+  stakingAmountSat: number;
+  stakingTimelock: number;
+}
+
+interface InclusionProof {
+  pos: number;
+  merkle: string[];
+  blockHash: string;
+}
```

Comment:
Would be nice to have comments on the above struct, e.g. the expected formats and what each entry represents.